backend/apps/ollama/main.py

```diff
@@ -713,25 +705,31 @@ class GenerateChatCompletionForm(BaseModel):
     keep_alive: Optional[Union[int, str]] = None
 
 
+def get_ollama_url(url_idx: Optional[int], model: str):
+    if url_idx is None:
+        if model not in app.state.MODELS:
+            raise HTTPException(
+                status_code=400,
+                detail=ERROR_MESSAGES.MODEL_NOT_FOUND(model),
+            )
+        url_idx = random.choice(app.state.MODELS[model]["urls"])
+    url = app.state.config.OLLAMA_BASE_URLS[url_idx]
+    return url
+
+
 @app.post("/api/chat")
 @app.post("/api/chat/{url_idx}")
 async def generate_chat_completion(
     form_data: GenerateChatCompletionForm,
     url_idx: Optional[int] = None,
     user=Depends(get_verified_user),
 ):
-
-    log.debug(
-        "form_data.model_dump_json(exclude_none=True).encode(): {0} ".format(
-            form_data.model_dump_json(exclude_none=True).encode()
-        )
-    )
+    log.debug(f"{form_data.model_dump_json(exclude_none=True).encode()}=")
 
     payload = {
         **form_data.model_dump(exclude_none=True, exclude=["metadata"]),
     }
-    if "metadata" in payload:
-        del payload["metadata"]
+    payload.pop("metadata", None)
```

Comment:
if "metadata" in payload:
    del payload["metadata"]

Reply:
![Screenshot 2024-08-08 at 10 09 29](https://github.com/user-attachments/assets/dd89acd3-db8a-4b8d-85c0-f886fd4ae283)

Reply:
`.pop("key", None)` works if the key is not present

Reply:
I believe `del` is slightly more efficient here, Let's put it back.

Reply:
undone